homeassistant/components/enphase_envoy/config_flow.py

```diff
@@ -89,24 +89,41 @@ async def async_step_zeroconf(
         self, discovery_info: zeroconf.ZeroconfServiceInfo
     ) -> ConfigFlowResult:
         """Handle a flow initialized by zeroconf discovery."""
+        current_hosts = self._async_current_hosts()
+        _LOGGER.debug(
+            "Zeroconf ip %s processing %s, current hosts: %s",
+            discovery_info.ip_address.version,
+            discovery_info.host,
+            current_hosts,
+        )
```

Comment:
It feels like we are missing test coverage here.  I think adding additional tests would be valuable as well.  There is also a chance you'll find the issue when writing the tests.

Reply:
I will look into adding more, but for now I also feel we should get this debug log out. The reports are pretty fuzzy and have little data on what actually is happening. I would like to prevent another month of delay to get this out and capture some more information, if any.

Reply:
Added some more test, no unexpected bahavior found.